src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LightCompiler.cs

```diff
@@ -108,7 +108,7 @@ internal bool HasHandler(InterpretedFrame frame, Exception exception, [NotNullWh
                 // Unreachable.
                 // Want to assert that this case isn't hit, but an assertion failure here will be eaten because
                 // we are in an exception filter. Therefore return true here and assert in the catch block.
-                handler = null;
+                handler = null!;
```

Comment:
:memo: the `[NotNullWhen(...)]` annotation on method seems incorrect

Reply:
See the comment above: "unreachable".

Reply:
Based on the comment this code block is unreachable.